include/admin/images.inc.php

```diff
@@ -322,12 +342,12 @@
 
                             // Insert into database
                             $image_id = serendipity_insertImageInDatabase($tfile, $serendipity['POST']['target_directory'][$tindex], $authorid, null, $realname);
+                            serendipity_plugin_api::hook_event('backend_image_add', $target, array('image_id' => $image_id));
```

Comment:
That completely changes the API event input parameters, why?

Reply:
That's a mixup on my side, this change is not supposed to be in this PR and it's' not made be me. I will revert it here. But iirc this change is a correction, this event is called two times with different parameters, this normalizes it.

Reply:
Hm ok, but I believe the proper code should contain the larger array, not just the one with "image_id"...?

Reply:
It really should not be in this PR. I will split this PR in a frontend/backend-PR, update smarty before that in master, also apply the UTF-8 fix there, and then re-apply my changes step by step. Then this change should vanish and the PRs be easier to review.